src/nodemetadata.cpp

```diff
@@ -101,14 +101,20 @@ void NodeMetadataList::serialize(std::ostream &os) const
 		v3s16 p = i->first;
 		NodeMetadata *data = i->second;
 
-		u16 p16 = p.Z * MAP_BLOCKSIZE * MAP_BLOCKSIZE + p.Y * MAP_BLOCKSIZE + p.X;
-		writeU16(os, p16);
-
+		if (uncompressed_pos) {
+			writeU16(os, p.X);
+			writeU16(os, p.Y);
+			writeU16(os, p.Z);
+		} else {
+			u16 p16 = p.Z * MAP_BLOCKSIZE * MAP_BLOCKSIZE + p.Y * MAP_BLOCKSIZE + p.X;
```

Comment:
@nerzhul 1. the calculations are quite lightweight as `MAP_BLOCKSIZE` is power of two, probably writing the u16 to the array would require more computation (putting the stuff to the array, incrementing the counter every time times), I guess compilers optimize this to a shift, and 2. changing this would require a change of the map format.